internal/email/sender.go

```diff
@@ -32,7 +32,7 @@ func NewEmailSender(conf *config.Config) *EmailSender {
 func (es *EmailSender) SendVerificationEmail(to, code string) error {
 	// msg := []byte(fmt.Sprintf("To: %s\r\nSubject: %s\r\n\r\n%s\r\n", to, subject, body))
 	msg := gomail.NewMessage()
-	msg.SetHeader("From", "no-reply@donetick.com")
+	msg.SetHeader("From", es.client.Username)
```

Comment:
I think we don't have `Username` in the email configuration. So I don't think this will compile. We can add a new attribute in the client `From` and use it to populate the template

Reply:
The NewDialer takes function takes the Username as an argument: https://pkg.go.dev/gopkg.in/gomail.v2#NewDialer
In our case, the email address in the config is the username:
```go
	client := gomail.NewDialer(conf.EmailConfig.Host, conf.EmailConfig.Port, conf.EmailConfig.Email, conf.EmailConfig.Key)
```
https://github.com/donetick/donetick/blob/b9eda3438aefa622d4c8b0cb7fb6d8d26ad08302/internal/email/sender.go#L20C29-L20C33

The code compiles and the FROM header takes the same values as `conf.EmailConfig.Email`.

I tested the code, as this is a feature I needed (for the password reset) for self-hosting donetick for our chores at home :)

Reply:
oh, that makes sense! Thank you for your contribution! It will be merge in the next release @Alone2